contracts/Zauction.sol

```diff
@@ -119,6 +119,23 @@ contract ZAuction is Initializable, OwnableUpgradeable {
     );
   }
 
+  /// Cancels an existing bid for an NFT by marking it as already consumed
+  /// so that it can never be fulfilled.
+  /// @param account The account that made the specific bid
+  /// @param auctionId The ID of the auction associated with that bid
+  function cancelBid(address account, uint256 auctionId) external {
+    require(
+      msg.sender == account,
+      "zAuction: Cannot cancel someone else's bid"
+    );
+    require(
+      !consumed[account][auctionId],
+      "zAuction: Cannot cancel an already consumed bid"
+    );
+
+    consumed[account][auctionId] = true;
```

Comment:
nope, honestly just still getting my head around the nitty-gritty of smart contract best practices. My intuition was that it might be handy to be able to determine whether a bid was consumed because it was accepted or cancelled after the fact from logs but it doesn't seem strictly necessary and [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).